Calendar/BookingsDiagnosticSummary.ps1

```diff
@@ -0,0 +1,170 @@
+$BuildVersion = ""
+. $PSScriptRoot\..\Shared\ScriptUpdateFunctions\Test-ScriptVersion.ps1
+if (Test-ScriptVersion -AutoUpdate -Confirm:$false) {
+    # Update was downloaded, so stop here.
+    Write-Host "Script was updated. Please rerun the command." -ForegroundColor Yellow
+    return
+}
+
+Write-Verbose "Script Versions: $BuildVersion"
+
+# ===================================================================================================
+# Helper scripts
+# ===================================================================================================
+. $PSScriptRoot\BookingHelpers\BookingGenericFunctions.ps1
+. $PSScriptRoot\BookingHelpers\BookingTenantSettingsHelper.ps1
+. $PSScriptRoot\BookingHelpers\BookingTenantSettingsLogic.ps1
+. $PSScriptRoot\BookingHelpers\BookingMBHelpers.ps1
+. $PSScriptRoot\BookingHelpers\BookingMBLogic.ps1
+. $PSScriptRoot\BookingHelpers\BookingMessageTrackingLogHelper.ps1
+. $PSScriptRoot\BookingHelpers\BookingMessageTrackingLogLogic.ps1
+. $PSScriptRoot\BookingHelpers\BookingStaffHelpers.ps1
+. $PSScriptRoot\BookingHelpers\BookingStaffLogic.ps1
+. $PSScriptRoot\BookingHelpers\bookingStaffLogHelper.ps1
+. $PSScriptRoot\BookingHelpers\bookingStaffLogLogic.ps1
+. $PSScriptRoot\BookingHelpers\fileSaveHelpers.ps1
+. $PSScriptRoot\BookingHelpers\ExcelWrite.ps1
+. $PSScriptRoot\CalLogHelpers\Write-DashLineBoxColor.ps1
+. $PSScriptRoot\CalLogHelpers\ExcelModuleInstaller.ps1
+
+
+
+
+# Populating Global Variables and getting general data
+
+# See if it is a Customer Tenant running the cmdlet. (They will not have access to Organization parameter)
+$script:MSSupport = [Bool](Get-Help Get-Mailbox -Parameter Organization -ErrorAction SilentlyContinue)
+Write-Verbose "MSSupport: $script:MSSupport"
+
+$script:PadCharsMessage = 40
+$script:indent = "         "
+$script:MessageTrackingDays = 5
+
+IsGetMailboxAvailable
+
+$script:Domain = SplitDomainFromEmail $identity -errorAction SilentlyContinue
+$script:OrgConfig=""
+$script:OWAMBPolicy = ""
+$script:AcceptedDomains = ""
+$script:TenantSettings = Get-BookingTenantSettings -Domain $script:Domain -ErrorAction SilentlyContinue
+
+#removed BMB variable, is unused
+#$script:BMB =""
+$script:bookingMBData = Get-BookingMBData -Identity $identity -ErrorAction SilentlyContinue
+$script:BookingStaffMembershipLog = GetStaffMembershipLogs -Identity $identity
+$script:BookingStaffMembershipLogArray = GetMembershipLogArray  -Identity $identity
+$script:MessageTrackingLogs = Get-MessageTrackingLogs -identity $identity -ErrorAction SilentlyContinue
+
+$script:MBPermissions = Get-MBPermissions -Identity $identity -ErrorAction SilentlyContinue
+$script:MBRecipientPermissions = Get-MBRecipientPermissions -Identity $identity -ErrorAction SilentlyContinue
+$script:StaffData = Get-StaffData | Format-Table
+
+
+
+
+$script:TenantSettings
+$script:bookingMBData
+$script:StaffData
+$script:MessageTrackingLogs
```

Comment:
This seem to be your main block of code getting data.
Do you want status messages?
i.e. Getting Mailbox...
Getting Booking Membership...

I guess it would depend on how long it takes.

Reply:
Nevermind, looks like each of the calls have the logging in them.